### PR TITLE
Add function to clean up existing models

### LINKED ISSUE #53: [FEATURE][docbot] Need function to clean up existing models after tests
### Is your feature request related to a problem?
There needs to be a function that can delete all of the items that were used when testing from connectors to indexes. This is a sample of the order that is needed to delete all the artifacts created when using a model.  

```
POST /_plugins/_ml/models/<MODEL_ID>/_undeploy
DELETE /_plugins/_ml/models/<MODEL_ID>
DELETE /_plugins/_ml/connectors/<CONNECTOR_ID>
DELETE _ingest/pipeline/cohere-ingest-pipeline
DELETE cohere-index
```

### PR DESCRIPTION
**Pull request by ortegaa32**
### Description
Adds a function to clean up existing models after testing.

### Issues Resolved
Resolves #53 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: demo/docbot/cluster_bootstrap.py [hunk 1]
```diff
@@ -293,6 +293,56 @@ def initialize_index(client: MLClient) -> None:
...
+  # Delete index
+  delete_index_response = client._client.indices.delete(index='cohere-index')
+  if delete_index_response:
+      print('\nDeleting index:')
+      print(delete_index_response)
+  else:
+      raise Exception("Failed to delete index.")
+
+  # Clear MODEL_STATE
+  for state in MODEL_STATE:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:344]: Here for clearing model state, lets instead take each of the parameters and assign their values to None. that way if there are multiple tests that depend on this data structure they can simply re-populate it.

**Review by dtaivpp — CHANGES_REQUESTED** [review]

**Comment by ortegaa32** [comment]
Just made the relevant change

File: demo/docbot/cluster_bootstrap.py [hunk 2]
```diff
@@ -293,6 +293,56 @@ def initialize_index(client: MLClient) -> None:
...
+  delete_pipeline_response = client._client.ingest.delete_pipeline(id='cohere-ingest-pipeline')
+  if delete_pipeline_response:
+      print('\nDeleting pipeline:')
+      print(delete_pipeline_response)
+  else:
+      raise Exception("Failed to delete ingest pipeline.")
+
+  # Delete index
+  delete_index_response = client._client.indices.delete(index='cohere-index')
+  if delete_index_response:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:337]: For index response, we should check for `{'acknowledged': True}` here

File: demo/docbot/cluster_bootstrap.py [hunk 3]
```diff
@@ -293,6 +293,56 @@ def initialize_index(client: MLClient) -> None:
...
+  delete_connector_response = client.delete_connector(MODEL_STATE['connector_id'])
+  if delete_connector_response:
+      print('\nDeleting connector:')
+      print(delete_connector_response)
+  else:
+      raise Exception("Failed to delete connector.")
+
+  # Delete ingest pipeline
+  delete_pipeline_response = client._client.ingest.delete_pipeline(id='cohere-ingest-pipeline')
+  if delete_pipeline_response:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:329]: Here we also want to check for `{'acknowledged': True}`

File: demo/docbot/cluster_bootstrap.py [hunk 4]
```diff
@@ -293,6 +293,56 @@ def initialize_index(client: MLClient) -> None:
...
+  delete_model_response = client.delete_model(MODEL_STATE["model_id"])
+  if delete_model_response:
+      print('\nDeleting model:')
+      print(delete_model_response)
+  else:
+      raise Exception("Failed to delete model.")
+
+  # Delete model connector
+  delete_connector_response = client.delete_connector(MODEL_STATE['connector_id'])
+  if delete_connector_response:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:321]: For deleting the connector we should look for result: deleted and failed: 0 in 

```json
{'_index': '.plugins-ml-connector', '_id': '63fKgYsBwPaemUsm3MGE', '_version': 2, 'result': 'deleted', '_shards': {'total': 1, 'successful': 1, 'failed': 0}, '_seq_no': 1, '_primary_term': 1}
```

File: demo/docbot/cluster_bootstrap.py [hunk 5]
```diff
@@ -293,6 +293,56 @@ def initialize_index(client: MLClient) -> None:
...
+  undeploy_model_response = client.undeploy_model(MODEL_STATE['model_id'])
+  if undeploy_model_response:
+      print('\nUndeploying model:')
+      print(undeploy_model_response)
+  else:
+      raise Exception("Failed to undeploy model.")
+
+  # Delete model
+  delete_model_response = client.delete_model(MODEL_STATE["model_id"])
+  if delete_model_response:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:313]: We should check here for result: deleted and failed: 0

```json
{'_index': '.plugins-ml-model', '_id': '8HfLgYsBwPaemUsmvMFi', '_version': 5, 'result': 'deleted', '_shards': {'total': 1, 'successful': 1, 'failed': 0}, '_seq_no': 4, '_primary_term': 1}
```

File: demo/docbot/cluster_bootstrap.py [hunk 6]
```diff
@@ -293,6 +293,56 @@ def initialize_index(client: MLClient) -> None:
...
+def ml_cleanup(client: MLClient) -> None:
+  """
+    Args:
+        client: MLCLient
+    Returns:
+        returns None
+    """
+  # Undeploy model
+  undeploy_model_response = client.undeploy_model(MODEL_STATE['model_id'])
+  if undeploy_model_response:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:305]: For this one we should check for stats: {<key1>: undeployed}

```json
{'sqGoQAqVQJGCzU6yLgTStg': {'stats': {'8HfLgYsBwPaemUsmvMFi': 'undeployed'}}}
```

**Review by dtaivpp — CHANGES_REQUESTED** [review]
Looks good over all! Just address the few comments and we are good to merge.

**Review by dtaivpp — APPROVED** [review]
Looks great thanks!

**Pull request merged**